timemachine/fe/rest/bond.py

```diff
@@ -0,0 +1,80 @@
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass, field
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+@dataclass(frozen=True)
+class CanonicalBond:
+    i: int
+    j: int
+
+    _unsafe: bool = field(default=False, init=True, compare=False, repr=False)
+
+    def __post_init__(self):
+        if not self._unsafe:
+            if self.i >= self.j:
+                raise ValueError(f"{(self.i, self.j)} is not canonical")
+
+    @classmethod
+    def from_idxs(cls, i: int, j: int):
+        return cls(i, j, _unsafe=True) if i < j else cls(j, i, _unsafe=True)
+
+    def translate(self, a_to_b: Mapping[int, int] | Sequence[int] | NDArray[np.int32]) -> "CanonicalBond":
+        return CanonicalBond.from_idxs(a_to_b[self.i], a_to_b[self.j])
+
+
+@dataclass(frozen=True)
+class CanonicalAngle:
+    i: int
+    j: int
+    k: int
+
+    _unsafe: bool = field(default=False, init=True, compare=False, repr=False)
+
+    def __post_init__(self):
+        if not self._unsafe:
+            if self.i >= self.k:
+                raise ValueError(f"{(self.i, self.j, self.k)} is not canonical")
+
+    @classmethod
+    def from_idxs(cls, i: int, j: int, k: int):
+        return cls(i, j, k, _unsafe=True) if i < k else cls(k, j, i, _unsafe=True)
+
+    def translate(self, a_to_b: Mapping[int, int] | Sequence[int] | NDArray[np.int32]) -> "CanonicalAngle":
+        return CanonicalAngle.from_idxs(a_to_b[self.i], a_to_b[self.j], a_to_b[self.k])
+
+
+@dataclass(frozen=True)
+class CanonicalTorsion:
```

Comment:
Add a comment this is for Proper torsions only

Reply:
Renamed to emphasize that this is intended for propers in https://github.com/proteneer/timemachine/pull/1424/commits/3290ffa526e0f007f5731323d8556756089215bb

(for impropers, we might want a similar definition that remembers which index corresponds to the central atom)